### PR TITLE
iNS: manufactured solution: test restart

### LINKED ISSUE #771: test restart capabilities
after #729 #731 #732 #734

### PR DESCRIPTION
**Pull request by richardschu**
fixes #771 (last part)

this tests for iNS manufactured solution restart involving:
for pure Dirichlet conditions:
.) ALE, moving grid
.) different $hp$ refinement
.) timesteppers: `BDFCoupledSolution` `BDFDualSplitting` `BDFPressureCorreection`
.) for `BDFDualSplitting` (bcs we want to reduce test volume where possible and this is redundant otherwise)

and then also $hp$ resolution changing and additionally:
`L2` -> `HDIV`
`HDIV` -> `L2`
in this case we need a periodic solution (no ALE here!)

~What is curious: `HDIV` has a higher error than `L2` in the latter case. I will comment the part.~
(this was due to an initial version using a no-periodic solution with periodic boundary conditions)

### CODE REVIEW

File: tests/restart/incompressible_navier_stokes.mpirun=2.output [hunk 1]
```diff
@@ -0,0 +1,2684 @@
...
+________________________________________________________________________________
+
+ Time step number = 33      t = 2.25000e-02 -> t + dt = 2.28125e-02
+________________________________________________________________________________
+
+Calculate error for velocity at time t = 2.5000e-02:
+  Relative error (L2-norm): 5.52249e-05
+
+Calculate error for pressure at time t = 2.5000e-02:
+  Relative error (L2-norm): 1.56751e-05
```
Comment (richardschu) [inline tests/restart/incompressible_navier_stokes.mpirun=2.output]: and get 5e-5 and 2e-5

so a bit is for sure the projection, BUT below we start with `HDIV`

**Review by richardschu — COMMENTED** [review]
The `HDIV` restart is consistent, but it is interesting that the error level increases so much. We should run convergence studies again to check if everything is alright.
@kronbichler did you fix anything in the other branch or is the `HDIV` there in principle the same solver?

I will update #368 to make our lives easier

File: tests/restart/incompressible_navier_stokes.mpirun=2.output [hunk 2]
```diff
@@ -0,0 +1,2684 @@
...
+
+ Time step number = 73      t = 2.25000e-02 -> t + dt = 2.28125e-02
+________________________________________________________________________________
+
+Calculate error for velocity at time t = 2.5000e-02:
+  Relative error (L2-norm): 4.50386e-05
+
+Calculate error for pressure at time t = 2.5000e-02:
+  Relative error (L2-norm): 1.22256e-05
+
```
[the remark this answers is not in the document]
Comment (kronbichler) [reply tests/restart/incompressible_navier_stokes.mpirun=2.output]: That's indeed a bit high. We should expect slightly larger errors because H(div) with degree 10/9 is less rich a space than DG with degree 10. I first thought it could be an issue with solver tolerances, because for the coupled solver for H(div) a different scaling compared to L2 is used, but then I saw you use dual splitting. But I see the generalized Newtonian solver is on, can you check again with the Newtonian one? Or at least the parameters such that we captured that part? I agree that we should aim to find the reason for this difference.

[the remark this answers is not in the document]
Comment (richardschu) [reply tests/restart/incompressible_navier_stokes.mpirun=2.output]: yeah I will push a version w Newtonian and single core next, let's see how that behaves.

Also, the test is only failing on github, I do not know yet what to make of it.

**Comment by richardschu** [comment]
https://github.com/exadg/exadg/pull/831/commits/dfe2c5529ef3579c9828f0d248593de00ffcf47b
now has the cases with HDIV only, hence `do_not_merge`

we do coupled solution first, then dual splitting
Newtonian only, also the restart uses the identical grid now.
Only single core.
This debugs at the same time why the test is failing at restart and also what is up with HDIV+dual splitting. 

Regarding the latter case, I will do convergence studies now.
I think HDIV and dual splitting is just not as good `per se`.
The high error has nothing to do with restart.
(But restart has to be fixed as well, it reads some wrong solution somehow when running on github)

**Comment by richardschu** [comment]
I will continue here as soon as `HDIV` is fixed.

**Comment by richardschu** [comment]
Move this test to `vortex_periodic_bc` as the `navier_stokes_manufactured` is not periodic.

**Review by kronbichler — APPROVED** [review]
